### ISSUE TITLE
layerdrop in Wav2Vec2Adapter

### ISSUE THREAD

**Issue by bofenghuang** [huggingface/transformers#20505]
### System Info

Hi!

As mentioned in https://github.com/huggingface/transformers/issues/20451, the layer dropout in `Wav2Vec2Adapter` may produce outputs with different lengths.

I understand the use of layerdrop in transformer structure, but do we need it in CNNs (`Wav2Vec2Adapter`)?

https://github.com/huggingface/transformers/blob/61d3928bfb3029bceb5be3e68ca3d4bf8456758f/src/transformers/models/wav2vec2/modeling_wav2vec2.py#L1006-L1009

### Who can help?

cc @anton-l @patrickvonplaten 

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

I copied the reproduction code of @OllieBroadhurst  in https://github.com/huggingface/transformers/issues/20451

```python
from transformers import Wav2Vec2Model

model = Wav2Vec2Model.from_pretrained("anton-l/wav2vec2-base-lang-id", 
                                      add_adapter=True, 
                                      adapter_stride=2,
                                      adapter_kernel_size=3, 
                                      num_adapter_layers=2)

model.train() # NB

dummy_input = torch.randn((1, 16000))

expected_output_sequence_length = 13

for _ in range(200):
    output_shape = model(input_values=dummy_input)[0].shape[1]
    if output_shape != expected_output_sequence_length:
        print(output_shape)
```

### Expected behavior

The above loop shouldn't print anything out.

**Comment by patrickvonplaten** [comment]
@sanchit-gandhi could you take a look here?

**Comment by bofenghuang** [comment]
Thanks @sanchit-gandhi !
